doozer/doozerlib/backend/konflux_client.py

```diff
@@ -712,3 +714,84 @@ def _inner():
                     raise
 
         return await exectools.to_thread(_inner)
+
+    async def wait_for_release(self, release_name: str, namespace: Optional[str] = None,
+                               overall_timeout_timedelta: Optional[datetime.timedelta] = None) -> resource.ResourceInstance:
+        """
+        Wait for a Release to complete.
+
+        :param release_name: The name of the Release.
+        :param namespace: The namespace of the Release.
+        :param overall_timeout_timedelta: Maximum time to wait for release to complete before exiting (defaults to 5 hour)
+        :return: The Release ResourceInstance
+        """
+        if overall_timeout_timedelta is None:
+            overall_timeout_timedelta = datetime.timedelta(hours=5)
```

Comment:
These hard-coded constants are always sneaky to find when we want to change them... maybe there's a better place for them?